packages/core/src/tools/tools.ts

```diff
@@ -231,6 +232,23 @@ export abstract class BaseDeclarativeTool<
     return this.createInvocation(params);
   }
 
+  override validateToolParams(params: TParams): string | null {
+    const errors = SchemaValidator.validate(
+      this.schema.parametersJsonSchema,
+      params,
+    );
+
+    if (errors) {
+      return errors;
+    }
+    return this.validateToolParamValues(params);
+  }
+
+  protected validateToolParamValues(_params: TParams): string | null {
```

Comment:
nit: why this pattern instead of requireing tools to call 
super.validateToolParams?

With this there is a risk someone will accidentally override validateToolParams instead causing problems.  Not sure if there is a lint like must call super in Dart to make that easy to enforce.

Reply:
We have `noImplicitOverrides` setup in our tsconfig, so overriding it unintentionally isn't completely trivial?

I'm not sure if there is a lint to call `super`, I can investigate this. Would you prefer it over the existing logic with `noImplicitOverride`?